Source/nthread.cpp

```diff
@@ -216,9 +216,23 @@ bool nthread_has_500ms_passed()
 {
 	int currentTickCount = SDL_GetTicks();
 	int ticksElapsed = currentTickCount - last_tick;
-	if (!gbIsMultiplayer && ticksElapsed > gnTickDelay * 10) {
-		last_tick = currentTickCount;
-		ticksElapsed = 0;
+	// Check if we missed multiple game ticks (> 10)
+	if (ticksElapsed > gnTickDelay * 10) {
+		bool resetLastTick = true;
+		if (gbIsMultiplayer) {
+			for (size_t i = 0; i < Players.size(); i++) {
+				if ((player_state[i] & PS_CONNECTED) != 0 && i != MyPlayerId) {
+					// Reset last tick is not allowed when other players are connected, cause the elapsed time is needed to sync the game ticks between the clients

```

Comment:
Note:
Locally I tested that the change also works with connected clients but I was unsure if my tests are really correct or if I missed something. That's why I only enabled it for solo multiplayer.
For the reported use-case (switching apps) fixing solo multiplayer should be good enough, cause in multiplayer with other clients there would be a timeout anyway.

Reply:
Yeah, I wouldn't expect this to have any real effect on non-solo multiplayer. The `nthread_recv_turns()` function updates `last_tick` as soon as the hourglass resolves.

https://github.com/diasurgical/devilutionX/blob/d0c1aba463fef3444e0a0be09dffc17f47ad43cb/Source/nthread.cpp#L121-L132

I feel like we can get rid of `sgbTicsOutOfSync` by replacing that logic with what you have here. What do you think?

Reply:
Wait, actually, the purpose of this might be to force `nthread_has_500ms_passed()` to return true on every frame so that `game_loop()` can rapidly check if the missing turns have arrived. I guess it might be a bit more complicated to replace the `nthread_recv_turns()` logic.

Reply:
I also was unsure that's why I didn't touch that part.
For now I think this pr is good enough, cause we don't have extreme speed ups in lag situations.
My thought was to get single- and multiplayer to behave similar when playing solo multiplayer.